### DISCUSSION TITLE
Should tar_group_by keep dplyr groups?

### DISCUSSION THREAD

**Discussion by kendonB** [ropensci/tarchetypes#53] [Ideas]
I am trying out `tar_group_by`. I unexpectedly found that the grouped target retains `dplyr` groups. I would have thought it would be best to lose those `dplyr` groups because the user isn't explicitly asking for them. Thoughts?

``` r
targets::tar_dir({ # tar_dir() runs code from a temporary directory.
  targets::tar_script({
    produce_data <- function() {
      expand.grid(var1 = c("a", "b"), var2 = c("c", "d"), rep = c(1, 2, 3))
    }
    list(
      tarchetypes::tar_group_by(data, produce_data(), var1, var2),
      tar_target(group, data, pattern = map(data))
    )
  })
  targets::tar_make()
  # dplyr groups are there
  dplyr::groups(targets::tar_read(data))
})
#> • start target data
#> • built target data
#> • start branch group_7f70acf9
#> • built branch group_7f70acf9
#> • start branch group_f28f3eed
#> • built branch group_f28f3eed
#> • start branch group_4f6b91f4
#> • built branch group_4f6b91f4
#> • start branch group_d878e3c0
#> • built branch group_d878e3c0
#> • built pattern group
#> • end pipeline
#> [[1]]
#> var1
#> 
#> [[2]]
#> var2
```

<sup>Created on 2021-05-20 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by wlandau** [comment]
That's an interesting point. Is there a situation where lingering `dplyr` groups would cause undesirable or unexpected behavior in terms of the important content of the data frames? I'm not sure myself because each branch has only one `dplyr` group. Related: groups are dropped in the `"fst_*"` formats (not sure about `"arrow"` and `"parquet"`, those at least hold on to class attributes).

**Reply by kendonB** [reply]
If the target is used as the full target, it will have different behavior to if it did not have groups. Forgetting your `data.frame` is grouped is a classic source of `dplyr` bugs. Given the behavior is also dependent on format, this is another motivation for removing dplyr groups.

**Reply by wlandau** [reply]
Should be fixed now via `targets::tar_group()`: https://github.com/ropensci/targets/commit/cdb5c1456b4bee91e3aba7babdd36fa2fa71218d
